coalib/results/AbsolutePosition.py

```diff
@@ -0,0 +1,55 @@
+from coalib.results.TextPosition import TextPosition
+from coalib.misc.Decorators import enforce_signature
+
+
+class AbsolutePosition(TextPosition):
+
+    @enforce_signature
+    def __init__(self,
+                 text: (tuple, list, None)=None,
+                 position: (int, None)=None):
+        """
+        Creates an AbsolutePosition object that represents the index of a
+        character in a string.
+
+        :param text:     The text containing the character.
+        :param position: Position identifying the index of character
+                         in text.
+        """
+        line = column = None
+        if position and text:
+            line, column = calc_line_col(text, position)
+        self._text = text
+        self._position = position
+        super().__init__(line, column)
+
+    @property
+    def position(self):
+        return self._position
+
+
+def calc_line_col(text, position):
```

Comment:
This function and the class above does the same thing. But class is named
and this function returns pair of numbers. It seems redundant to have both.
While inproducing to the class properties line and column will make code
more readable in the future.

0:41, ср, 06.04.2016 Lasse Schuirmann <notifications@github.com>:

> In coalib/results/AbsolutePosition.py
> <https://github.com/coala-analyzer/coala/pull/1947#discussion_r58618574>:
>
> > +        :param position: Position identifying the index of character
> > +                         in text.
> > +        """
> > +        line = column = None
> > +        if position and text:
> > +            line, column = calc_line_col(text, position)
> > +        self._text = text
> > +        self._position = position
> > +        super().__init__(line, column)
> > +
> > +    @property
> > +    def position(self):
> > +        return self._position
> > +
> > +
> > +def calc_line_col(text, position):
>
> will this functionality is general purpose and shoule be outside the class
> IMO, better name might be possible though
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala/pull/1947/files/cf89f1ec8ad4acc3432a4bc1c9e235c63c12d807#r58618574>
>